### PR TITLE
move cli main into `lib/`

### PR DESCRIPTION
**Pull request by pq**
Fixes: #1102.

/cc @bwilkerson @grouma @a14n

### CODE REVIEW

File: lib/src/cli.dart [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+  var customPackageRoot = options['package-root'];
+  if (customPackageRoot != null) {
+    lintOptions.packageRootPath = customPackageRoot;
+  }
+
+  var packageConfigFile = options['packages'];
+
+  if (customPackageRoot != null && packageConfigFile != null) {
+    errorSink.write("Cannot specify both '--package-root' and '--packages'.");
+    exitCode = unableToProcessExitCode;
```
Comment (kevmoo) [inline lib/src/cli.dart]: pro tip: return the exit code from this method – set it in `bin`. Makes testing MUCH easier.

File: lib/src/cli.dart [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+        fileCount: linter.numSourcesAnalyzed,
+        fileRoot: commonRoot,
+        showStatistics: stats,
+        machineOutput: options['machine'],
+        quiet: options['quiet'])
+      ..write();
+    // ignore: avoid_catches_without_on_clauses
+  } catch (err, stack) {
+    errorSink.writeln('''An error occurred while linting
+  Please report it at: github.com/dart-lang/linter/issues
```
Comment (kevmoo) [inline lib/src/cli.dart]: Move this catch to `bin` – makes testing easier

Comment (kevmoo) [reply lib/src/cli.dart]: ...make sure to set the exit code to an error!

File: lib/src/cli.dart (hunk 1 above)
Comment (pq) [reply lib/src/cli.dart]: Great point.  That said, I think these cases are pretty well covered in our integration tests.  Are you making a general point or do you see cases we're missing?

**Comment by kevmoo** [comment]
General point – I find it much easier to reason about code that way.

**Pull request merged**